From 3076ec10d6180199a8c1328cff688b2a6f52c607 Mon Sep 17 00:00:00 2001
From: Dinh Nguyen <dinguyen@altera.com>
Date: Tue, 16 Oct 2012 11:02:35 -0600
Subject: [PATCH 014/142] arm: socfpga: Enable spi and qspi driver for socfpga

-Add spi and qspi driver device tree entries
-Add spi and qspi into socfpga_defconfig

Signed-off-by: Dinh Nguyen <dinguyen@altera.com>
[Original patch taken from
 https://github.com/altera-opensource/linux-socfpga.git
 tag rel_socfpga-4.8_17.03.01_pr]
Signed-off-by: Yanjiang Jin <yanjiang.jin@windriver.com>
---
 arch/arm/boot/dts/socfpga.dtsi     | 34 +++++++++++++++++++++++++++-------
 arch/arm/configs/socfpga_defconfig |  7 +++++++
 2 files changed, 34 insertions(+), 7 deletions(-)

diff --git a/arch/arm/boot/dts/socfpga.dtsi b/arch/arm/boot/dts/socfpga.dtsi
index 9f48141270b8..512aa8cee1d1 100644
--- a/arch/arm/boot/dts/socfpga.dtsi
+++ b/arch/arm/boot/dts/socfpga.dtsi
@@ -729,25 +729,45 @@
 		};
 
 		spi0: spi@fff00000 {
-			compatible = "snps,dw-apb-ssi";
+			compatible = "snps,dw-spi-mmio";
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0xfff00000 0x1000>;
 			interrupts = <0 154 4>;
-			num-cs = <4>;
-			clocks = <&spi_m_clk>;
+			num-chipselect = <4>;
+			bus-num = <0>;
+			tx-dma-channel = <&pdma 16>;
+			rx-dma-channel = <&pdma 17>;
+			clocks = <&per_base_clk>;
 			status = "disabled";
+
+			spidev@0 {
+				compatible = "spidev";
+				reg = <0>;	/* chip select */
+				spi-max-frequency = <100000000>;
+				enable-dma = <1>;
+			};
 		};
 
 		spi1: spi@fff01000 {
-			compatible = "snps,dw-apb-ssi";
+			compatible = "snps,dw-spi-mmio";
 			#address-cells = <1>;
 			#size-cells = <0>;
 			reg = <0xfff01000 0x1000>;
-			interrupts = <0 155 4>;
-			num-cs = <4>;
-			clocks = <&spi_m_clk>;
+			interrupts = <0 156 4>;
+			num-chipselect = <4>;
+			bus-num = <1>;
+			tx-dma-channel = <&pdma 20>;
+			rx-dma-channel = <&pdma 21>;
+			clocks = <&per_base_clk>;
 			status = "disabled";
+
+			spidev@0 {
+				compatible = "spidev";
+				reg = <0>;
+				spi-max-frequency = <100000000>;
+				enable-dma = <1>;
+			};
 		};
 
 		sysmgr: sysmgr@ffd08000 {
diff --git a/arch/arm/configs/socfpga_defconfig b/arch/arm/configs/socfpga_defconfig
index 9f84be5b3ac5..fe7784b824b2 100644
--- a/arch/arm/configs/socfpga_defconfig
+++ b/arch/arm/configs/socfpga_defconfig
@@ -85,6 +85,13 @@ CONFIG_GPIO_DWAPB=y
 CONFIG_PMBUS=y
 CONFIG_SENSORS_LTC2978=y
 CONFIG_SENSORS_LTC2978_REGULATOR=y
+CONFIG_SPI=y
+CONFIG_SPI_CADENCE_QSPI=y
+CONFIG_SPI_DESIGNWARE=y
+CONFIG_SPI_DW_PL330_DMA=y
+CONFIG_SPI_DW_MMIO=y
+CONFIG_SPI_SPIDEV=y
+# CONFIG_RTC_HCTOSYS is not set
 CONFIG_WATCHDOG=y
 CONFIG_DW_WATCHDOG=y
 CONFIG_MFD_STMPE=y
-- 
2.11.0

